src/cryptoadvance/specter/util/psbt_creator.py

```diff
@@ -0,0 +1,183 @@
+                raise SpecterError(
+                    "Probably you don't have enough funds, or something else..."
+                )
+            else:
+                # calculate new amount if we need to subtract
+                if self.kwargs["subtract"]:
+                    for v in self.psbt["tx"]["vout"]:
+                        if self.addresses[0] in v["scriptPubKey"].get(
+                            "addresses", [""]
+                        ) or self.addresses[0] == v["scriptPubKey"].get("address", ""):
+                            self.amounts[0] = v["value"]
+        except Exception as e:
+            logger.exception(e)
+            raise SpecterError(f"{e} ... check the logs for the stacktrace")
+
+    @classmethod
+    def paymentinfo_from_ui(cls, specter, wallet, request_form):
+        """calculates the correct format needed by wallet.createpsbt() out of a request-form
+        returns something like  (addresses, labels, amounts, amount_units) (all arrays)
+        """
+        i = 0
+        addresses = []
+        labels = []
+        amounts = []
+        amount_units = []
+        while "address_{}".format(i) in request_form:
+            logger.debug(f"ssss {request_form['address_0']}")
+            addresses.append(request_form["address_{}".format(i)])
+            amount = 0.0
+            try:
+                amount = float(request_form["btc_amount_{}".format(i)])
+            except ValueError:
+                pass
+            if isnan(amount):
+                amount = 0.0
+            amounts.append(amount)
+            unit = request_form["amount_unit_{}".format(i)]
+            if specter.is_liquid and unit in ["sat", "btc"]:
+                unit = specter.default_asset
+            amount_units.append(unit)
+            labels.append(request_form["label_{}".format(i)])
+            if request_form["label_{}".format(i)] != "":
+                wallet.setlabel(addresses[i], labels[i])
+            i += 1
+        return addresses, labels, amounts, amount_units
+
+    @classmethod
+    def paymentinfo_from_text(
+        cls, specter, wallet, recipients_txt, recipients_amount_unit
+    ):
+        """calculates the correct format needed by wallet.createpsbt() out of a request-form
+        out of a textbox holding addresses and amounts.
+        """
+        i = 0
+        addresses = []
+        labels = []
+        amounts = []
+        amount_units = []
+        for output in recipients_txt.splitlines():
+            addresses.append(output.split(",")[0].strip())
+            if recipients_amount_unit == "sat":
+                amounts.append(float(output.split(",")[1].strip()) / 1e8)
+            elif recipients_amount_unit == "btc":
+                amounts.append(float(output.split(",")[1].strip()))
+            else:
+                raise SpecterError(
+                    f"Unknown recipients_amount_unit: {recipients_amount_unit}"
+                )
+            labels.append("")
```

Comment:
```suggestion
            labels.append("")
        return addresses, labels, amounts, amount_units
```

Missing a return here so currently the text option crash